context.go

```diff
@@ -402,12 +404,17 @@ func (ctx Context) LoadModuleByID(id string, rawMsg json.RawMessage) (any, error
 	// module has been configured for DNS challenges)
 	if appModule, ok := val.(App); ok {
 		ctx.cfg.apps[id] = appModule
+		defer func() {
+			if err != nil {
+				delete(ctx.cfg.apps, id)
+			}
+		}()
```

Comment:
As said in this [comment](https://github.com/caddyserver/caddy/pull/7070#issuecomment-3126028802), recording the error of an app that fails to provision or validate will only initialize the app once and reduce the work caddy needs to do.